modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx

```diff
@@ -441,165 +295,14 @@ const Actions = ({fdsView, namespace, spritemap}: IFDSViewSectionInterface) => {
-										size={4}
-									>
-										<ClayIcon
-											className="mr-4"
-											symbol={iconSymbol}
-										/>
-
-										<ClayForm.Group>
-											<label htmlFor={iconFormElementId}>
-												{Liferay.Language.get('icon')}
-											</label>
-
-											<ClaySelectWithOption
-												id={iconFormElementId}
-												onChange={(event) =>
-													setIconSymbol(
-														event.target.value
-													)
-												}
-												options={availableIconSymbols}
-												value={iconSymbol}
-											/>
-										</ClayForm.Group>
-									</ClayLayout.Col>
-								</ClayLayout.Row>
-							</ClayPanel.Body>
-						</ClayPanel>
-
-						<ClayPanel
-							collapsable
-							defaultExpanded
-							displayTitle={Liferay.Language.get(
-								'action-behavior'
-							)}
-						>
-							<ClayPanel.Body>
-								<ClayLayout.Row justify="start">
-									<ClayLayout.Col size={4}>
-										<ClayForm.Group>
-											<label htmlFor={typeFormElementId}>
-												{Liferay.Language.get('type')}
-
-												<RequiredMark />
-											</label>
-
-											<ClaySelectWithOption
-												id={typeFormElementId}
-												onChange={(event) =>
-													setType(event.target.value)
-												}
-												options={TYPES}
-												placeholder={Liferay.Language.get(
-													'please-select-an-option'
-												)}
-												value={type}
-											/>
-										</ClayForm.Group>
-									</ClayLayout.Col>
-								</ClayLayout.Row>
-
-								<ClayLayout.Row justify="start">
-									<ClayLayout.Col lg>
-										<ClayForm.Group>
-											<label htmlFor={urlFormElementId}>
-												{Liferay.Language.get('url')}
-
-												<RequiredMark />
-											</label>
-
-											<ClayInput
-												component="textarea"
-												id={urlFormElementId}
-												onChange={(event) =>
-													setURL(event.target.value)
-												}
-												placeholder={Liferay.Language.get(
-													'add-a-url-here'
-												)}
-												value={url}
-											/>
-										</ClayForm.Group>
-
-										<ConfirmationMessageRow />
-									</ClayLayout.Col>
-								</ClayLayout.Row>
-							</ClayPanel.Body>
-						</ClayPanel>
-
-						<ClayButton.Group className="pb-4 px-4" spaced>
-							<ClayButton
-								disabled={saveButtonDisabled}
-								onClick={saveFDSAction}
-							>
-								{Liferay.Language.get('save')}
-							</ClayButton>
-
-							<ClayButton
-								displayType="secondary"
-								onClick={() =>
-									setActiveSection(SECTIONS.ACTIONS)
-								}
-							>
-								{Liferay.Language.get('cancel')}
-							</ClayButton>
-						</ClayButton.Group>
-					</>
+					<ActionCreationForm
+						fdsView={fdsView}
+						getFDSActions={getFDSActions}
+						namespace={namespace}
+						sections={SECTIONS}
+						setActiveSection={setActiveSection}
+						spritemap={spritemap}
+					/>
```

Comment:
Splitting components into separate ones, even in separate files, definitely makes sense! Indent depth and line splitting is getting out of hand in this file. I also like the separate of state scope, there is no reason for actions list to be under form-specific states :+1: 

Couple of nits here:
- `js/components` dir does not seem right for the new file. It contains common inner components of multiple components. This one is very specific, tied to actions. The proper place for this would be either next to parent component, or in subdir, something like (`js/fds_view/actions/ActionCreationForm`).
- we should call this something like `ItemActionCreationForm`, `NewItemActionSection` or `NewItemAction`. The `Item` part in name is important, to prepare for future variant with create actions.
- we might want to further split this component down the line, to include different components for other item actions types (`async`, `modal` etc), but it's probably fine now the way it is.
- for symmetry, it might make sense to also extract the other section into a component next to the form one. Something like `ActionsList`. It's not necessary though.

Reply:
As a heads up I'm also re-using this form for the edit action with an added `initialValues` prop (https://github.com/liferay-frontend/liferay-portal/pull/3654), so it might make sense to not use the words `Creation`/`New`.  

Maybe simply `ItemAction`? This would also match up with the other section names like `FDSView`, `Details`, etc.

Reply:
I'd say `ItemAction`, `ItemActionForm` or `ItemActionSection`. Choose your destiny! :slightly_smiling_face: